src/NHibernate/Hql/Ast/ANTLR/Tree/DotNode.cs

```diff
@@ -417,8 +415,15 @@ private void DereferenceEntity(EntityType entityType, bool implicitJoin, string
 				               || comparisonWithNullableEntity;
 			}
 
-			if ( joinIsNeeded ) 
+			if ( joinIsNeeded )
 			{
+				//TODO: Find correct place for this if.
+				if (Walker.IsNullComparison)
```

Comment:
Need to find correct place for this check. It probably needs to be done somewhere earlier. I don't have time right now, could look later, but not promising it.